helix-view/src/document.rs

```diff
@@ -192,6 +201,14 @@ impl Document {
 
         async move {
             use tokio::{fs::File, io::AsyncWriteExt};
+            if let Some(parent) = path.parent() {
+                // TODO: display a prompt asking the user if the directories should be created
+                if !parent.exists() {
+                    return Err(Error::msg(
+                        "can't save file, parent directory does not exist",
+                    ));
+                }
+            }
             let mut file = File::create(path).await?;
```

Comment:
Wait, doesn't this already create the file? Currently it already does that.

Reply:
It does create the file, that’s correct

Reply:
I’m not sure what you’re getting at, could you explain the question further?

Reply:
I’m also thinking there should probably be an explicit error type so it can later be handled for example to show the prompt asking the user if the directories should be created.